aggregate-limitations.rst

```diff
@@ -0,0 +1,41 @@
+========================
+
+- Push of data to the server should be serialized. Within ODK Aggregate, there is a global mutex (`TaskLock` across all server instances, mediated by the datastore layer) in the server when inserting submissions. Having multiple push requests occurring simultaneously will cause them to block on the mutex, chewing up their 60-second request limit, as they get processed in single file no matter how many server instances are spun up.
+
+- The 60-second request limit can be very commonly exceeded over low-bandwidth connections, and even text-only submissions can be impacted on satellite connections. That is why ODK Collect splits submissions into multiple 10MB submission requests. The timer starts upon receipt of the first byte, so a slow connection can account for a sizeable portion of those 60 seconds. Same applies for sending a response. The processing on the server is generally negligible in relation to the transmission times.
+
+.. note::
+   
+   - The above two limitations, the global mutex and the in-memory copies/full-packet-assembly, are a result of implementing on top of AppEngine and its Datastore.
+   - A server that used database transactions and that used streaming servlet 3.0 functionality would have less trouble with concurrent requests.
+
+
+- Pull of data can be overlapped. However, for submissions with many media attachments, keep in mind that ODK Aggregate holds all media attachments in memory, then copies them into the HttpEntity, so this can consume a fair amount of memory (everything is held in memory at least twice). The same memory consumption applies on pushes / submission of forms. i.e., you probably want to manage how many media attachment download requests you have in play at the same time.
+
+- Spinning up of copies of the frontend will incur faster quota usage on AppEngine. For that reason, the Aggregate configuration here specifies a 14-second queuing time threshold before a new instance is spun up. Only if at least one request is queued for longer than 14 seconds will a new instance be spun up, and then that new instance will take about 30 seconds to become live. Leaving a 15-second processing interval. This is why ODK Collect tried twice before failing a submit.
+
+
+- Aggregate 0.9.x has a 1000-record limit to the number of records it can display (this is by design); it does, however, still retain all records i.e. it will hold as many records as you've uploaded and does not loose or ignore the 1001st record, etc; in general, any webserver will have a limit to what it can display interactively (at some point it will run out of memory or time out). 
+
+.. tip::
+  
+  - On Aggregate 0.9.x, we provide the Briefcase applet (one of the links on the top row) to allow you to extract your data from the server into a local CSV file for local processing. That should retrieve all data within the server, and should work regardless of the number of submissions you have uploaded.
```

Comment:
where is the link to the Briefcase applet?

Reply:
@adammichaelwood I meant to say here that the server shows the link at the top.
![selection_001](https://user-images.githubusercontent.com/15226041/31029690-559381fa-a570-11e7-89a0-6def8c124d82.png)
Do I need to insert the link here?

Reply:
Putting the link here in the docs would be a good idea.

Reply:
The link is an internal link. It opens up an ODK Briefcase page and generates a token. I think it is better to insert an image here.
![briefcase-applet](https://user-images.githubusercontent.com/15226041/31042916-3db42728-a5d0-11e7-9e6c-3cea0b345f11.png)

Reply:
@adammichaelwood Should I insert the image ?

Reply:
yes